.github/workflows/060_generate_benchmark_weights.yml

```diff
@@ -0,0 +1,73 @@
+name: Generate benchmark weights
+
+on:
+  workflow_dispatch:
+
+jobs:
+  benchmark-weights:
+    runs-on: [self-hosted, tfchainrunner01]
+    container:
+      image: threefolddev/tfchain:4
+      env:
+        DEBIAN_FRONTEND: noninteractive
+        PATH: /root/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin
+        RUSTUP_HOME: /root/.rustup
+        CARGO_HOME: /root/.cargo
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Chown user
+        run: |
+          chown -R $USER:$USER $GITHUB_WORKSPACE
+
+      - name: Cache build
+        uses: actions/cache@v3
+        timeout-minutes: 6
+        continue-on-error: true
+        with:
+          path: |
+            /root/.cargo/bin/
+            /root/.cargo/registry/index/
+            /root/.cargo/registry/cache/
+            /root/.cargo/git/db/
+            substrate-node/target/
+          key: ${{ runner.os }}-tfchain-build-cache-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-tfchain-build-cache
+
+      - name: Build
+        run: |
+          cd substrate-node
+          cargo build --release --features runtime-benchmarks
+
+      - name: Remove old weight files
+        run: |
+          rm substrate-node/pallets/*/src/weights.rs
+          
+      - name: Run benchmarking script
+        run: |
+          cd substrate-node/.maintain
+          chmod +x ./benchmark.sh
+          ./benchmark.sh
+
+      - name: Rebuild
```

Comment:
I’m not entirely sure what the benefit of rebuilding is in this context. Could you please clarify?

Reply:
Should this be handled by the existing Build and Test workflow? It is already configured to run with every push to PRs.
Or do you need to guard against pushing the commit if things do not compile? although it seems unlikely for these benchmarks to break the build but I’m not entirely sure.